client/consensus/grandpa/authorities.go

```diff
@@ -0,0 +1,814 @@
+	defer sas.mtx.Unlock()
+	return sas.inner.pendingChanges()
+}
+
+// currentLimit Get the earliest limit-block number, if any. If there are pending changes across
+// different forks, this method will return the earliest effective number (across the
+// different branches) that is higher or equal to the given min number.
+//
+// Only standard changes are taken into account for the current
+// limit, since any existing forced change should preclude the voter from voting.
+func (sas *SharedAuthoritySet[H, N]) currentLimit(min N) (limit *N) {
+	sas.mtx.Lock()
+	defer sas.mtx.Unlock()
+	return sas.inner.currentLimit(min)
+}
+
+func (sas *SharedAuthoritySet[H, N]) applyForcedChanges(bestHash H, bestNumber N, isDescendentOf IsDescendentOf[H], telemetry *Telemetry) (newSet *appliedChanges[H, N], err error) {
+	sas.mtx.Lock()
+	defer sas.mtx.Unlock()
+	return sas.inner.applyForcedChanges(bestHash, bestNumber, isDescendentOf, telemetry)
+}
+
+// ApplyStandardChanges Apply or prune any pending transitions based on a finality trigger. This
+// method ensures that if there are multiple changes in the same branch,
+// finalizing this block won't finalize past multiple transitions (i.e.
+// transitions must be finalized in-order). The given function
+// `is_descendent_of` should return `true` if the second hash (target) is a
+// descendent of the first hash (base).
+//
+// When the set has changed, the return value will be a Status type where newSetBlockInfo
+// is the canonical block where the set last changed (i.e. the given
+// hash and number).
+func (sas *SharedAuthoritySet[H, N]) ApplyStandardChanges(finalizedHash H, finalizedNumber N, isDescendentOf IsDescendentOf[H], telemetry *Telemetry) (Status[H, N], error) {
+	sas.mtx.Lock()
+	defer sas.mtx.Unlock()
+	return sas.inner.applyStandardChanges(finalizedHash, finalizedNumber, isDescendentOf, telemetry)
+}
+
+// EnactsStandardChange Check whether the given finalized block number enacts any standard
+// authority set change (without triggering it), ensuring that if there are
+// multiple changes in the same branch, finalizing this block won't
+// finalize past multiple transitions (i.e. transitions must be finalized
+// in-order). Returns *true if the block being finalized enacts a
+// change that can be immediately applied, *false if the block being
+// finalized enacts a change but it cannot be applied yet since there are
+// other dependent changes, and nil if no change is enacted. The given
+// function `is_descendent_of` should return `true` if the second hash
+// (target) is a descendent of the first hash (base).
+func (sas *SharedAuthoritySet[H, N]) EnactsStandardChange(finalizedHash H, finalizedNumber N, isDescendentOf IsDescendentOf[H]) (*bool, error) {
+	sas.mtx.Lock()
+	defer sas.mtx.Unlock()
+	return sas.inner.EnactsStandardChange(finalizedHash, finalizedNumber, isDescendentOf)
+}
+
+// Status of the set after changes were applied.
+type Status[H comparable, N constraints.Unsigned] struct {
```

Comment:
Don't think this needs to be public.  It's defined as `pub(crate)` in the rust code.